src/catalog/src/durable/upgrade.rs

```diff
@@ -252,52 +253,50 @@ pub(crate) async fn upgrade(
         version = new_version;
     }
 
-    /// Determines which upgrade to run for the `version` and executes it.
-    ///
-    /// Returns the new version and upper.
-    async fn run_upgrade(
-        unopened_catalog_state: &mut UnopenedPersistCatalogState,
-        mode: Mode,
-        version: u64,
-    ) -> Result<u64, CatalogError> {
-        let incompatible = DurableCatalogError::IncompatibleDataVersion {
-            found_version: version,
-            min_catalog_version: MIN_CATALOG_VERSION,
-            catalog_version: CATALOG_VERSION,
-        }
-        .into();
+    Ok(())
+}
 
-        match version {
-            ..=TOO_OLD_VERSION => Err(incompatible),
+/// Determines which upgrade to run for the `version` and executes it.
+///
+/// Returns the new version and upper.
+async fn run_upgrade(
+    unopened_catalog_state: &mut UnopenedPersistCatalogState,
+    mode: Mode,
+    version: u64,
+) -> Result<u64, CatalogError> {
+    let incompatible = DurableCatalogError::IncompatibleDataVersion {
+        found_version: version,
+        min_catalog_version: MIN_CATALOG_VERSION,
+        catalog_version: CATALOG_VERSION,
+    }
+    .into();
 
-            48 => {
-                run_versioned_upgrade(unopened_catalog_state, mode, version, v48_to_v49::upgrade)
-                    .await
-            }
-            49 => {
-                run_versioned_upgrade(unopened_catalog_state, mode, version, v49_to_v50::upgrade)
-                    .await
-            }
-            50 => {
-                run_versioned_upgrade(unopened_catalog_state, mode, version, v50_to_v51::upgrade)
-                    .await
-            }
-            51 => {
-                run_versioned_upgrade(unopened_catalog_state, mode, version, v51_to_v52::upgrade)
-                    .await
-            }
-            52 => {
-                run_versioned_upgrade(unopened_catalog_state, mode, version, v52_to_v53::upgrade)
-                    .await
-            }
+    match version {
+        ..=TOO_OLD_VERSION => Err(incompatible),
```

Comment:
Not tested according to coverage information (based on the tests pipeline). Is this plausible? Though, I think it should be tested by the upgrade tests in Nightly, right?

Reply:
This error should be unreachable except for programmer error. 

In order to test this code we would need to start Materialize on some version `x` s.t. the catalog version is `v`. Then directly upgrade materialize to version `y` s.t. the migration from catalog version `v` is removed. However, the check for skipping Materialize versions would be triggered before this check.